sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -25,17 +25,26 @@ import org.apache.spark.sql.catalyst.util.TypeUtils
 import org.apache.spark.sql.types._
 import org.apache.spark.util.Utils
 
+trait PredResult
```

Comment:
This will be needed for predicates in join / filter. Since previously, we only cares about the `true`, and will ignore the null value and the `false`, but for now, in `anti join`, we only cares about the `false`, and ignore the `null` or `true`, that's the motive I changed the interface of predicates from `(InternalRow => Boolean)` => `(InternalRow => PredResult)`.